### PR TITLE
Add Prettier configuration

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Adds the necessary configurations for configuring Prettier. And setting up a pre-commit hook for automatically running it before commits.

* Removed a couple of tslint rules that conflicts with prettier.
* Temporarily disable tslint in CI. This will be enabled again in the 2nd PR.
* Set `indent_size: 2`, since we decided to stick with prettiers defaults this should ensure our editors don't get confused. (While at it, I also added `trim_trailing_whitespace: true`
* Added huskier with a pre-commit hook relying on lint-staged. Which will ensure each commit is formatted. Note, this "should" never require manual intervention and will just make the commits easier to read.
* Installed prettier, with the only config change to set line-width to 100.

## Before you submit
- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -51,8 +51,8 @@ jobs:
       - name: Install Node dependencies
         run: npm install
 
-      - name: Run linter
-        run: npm run lint
+      # - name: Run linter
+      #   run: npm run lint
```
Comment (djsmith85) [inline .github/workflows/build.yml:55]: This is temporary as we'll be adding this step again once we've migrated to ESLint?

**Review by djsmith85 — COMMENTED** [review]
Looking good. The git-blame-ignore-revs will follow once jslib is done?

File: .github/workflows/build.yml (hunk 1 above)
Comment (Hinton) [reply .github/workflows/build.yml:55]: We're adding it back in the prettier PR.

**Comment by Hinton** [comment]
Yes.

**Review by djsmith85 — APPROVED** [review]
Looking good

**Pull request merged**